DPB_converger.py

```diff
@@ -0,0 +1,363 @@
+    config = configparser.ConfigParser()
+    config.sections()
+    config.read(config_path)
+    """
+    The script expects the files precision and exp_noise_rate to be in the folder
+    \PreAnalysisParameter\parameters, unlike the write_swift_batch.py which allows for a directory to be specified.
+    However this is where BatchPreAnalysis places them.
+    """
+    try:
+        if len([key for key in config["INPUT_DIRS"]]):
+            for key in config["INPUT_DIRS"]:
+                file_paths_dir = get_loc_files(config["INPUT_DIRS"][key] + "\\cells\\tracks")
+                n_file_paths.append(len(file_paths_dir))
+                file_paths.extend(file_paths_dir)
+                precision_vals.append(
+                    config["INPUT_DIRS"][key] + '\\PreAnalysisParameters\\parameters\\precisions.txt')
+                assert (len(n_file_paths) == len(precision_vals))
+                exp_noise_rate_vals.append(
+                    config["INPUT_DIRS"][
+                        key] + '\\PreAnalysisParameters\\parameters\\exp_noise_rate.txt')
+                assert (len(n_file_paths) == len(exp_noise_rate_vals))
+        else:
+            raise IncorrectConfigException("No input directory defined in config.")
+    except KeyError:
+        raise IncorrectConfigException("Section INPUT_DIRS missing in config.")
+
+    try:
+        tau = config["PARAMETERS_GLOBAL"]["tau"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter tau missing in config.")
+
+    try:
+        max_displacement = config["PARAMETERS_GLOBAL"]["max_displacement"]
+        max_displacement = str(float(max_displacement) * float(exp_displacement))
+    except KeyError:
+        raise IncorrectConfigException("Parameter max_displacement missing in config.")
+
+    try:
+        max_displacement_pp = config["PARAMETERS_GLOBAL"]["max_displacement_pp"]
+        max_displacement_pp = str(float(max_displacement_pp) * float(exp_displacement))
+    except KeyError:
+        raise IncorrectConfigException("Parameter max_displacement_pp missing in config.")
+
+    try:
+        p_switch = config["PARAMETERS_GLOBAL"]["p_switch"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter p_switch missing in config.")
+
+    try:
+        diffraction_limit = config["PARAMETERS_GLOBAL"]["diffraction_limit"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter diffraction_limit missing in config.")
+
+    try:
+        batch_path = config["SAVE_DIRECTORY"]["save_path"] + '\\swift.bat'
+    except KeyError:
+        raise IncorrectConfigException("No save directory defined in config.")
+
+    precision = []
+    for i in range(len(n_file_paths)):
+        if os.path.exists(precision_vals[i]):
+            precision.extend(pd.read_csv(precision_vals[i], sep=" ", encoding="latin").iloc[:, 1].to_list())
+        else:
+            precision += [precision_vals[i]] * n_file_paths[i]
+
+    exp_noise_rate = []
```

Comment:
add inline comment

Reply:
Those lines are from your code and to be honest, I am unsure why it is written that way. In my opinion it would make more sense to raise an exception instead of this.